driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -11,19 +15,77 @@ use futures::{
 };
 use orderbook::Orderbook;
 use std::future::Future;
-use std::sync::{Arc, Mutex};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex,
+};
+
+/// An event based orderbook that automatically updates itself with new events from the contract.
+#[derive(Debug)]
+pub struct UpdatingOrderbook {
+    orderbook: Arc<Mutex<Orderbook>>,
```

Comment:
Nit: I wonder if this could be an Option which is None initially and then the thread sends backs the orderbook on a one shot channel as soon as it's ready. It's semantically the same, but avoids having a boolean which is implicitly coupled to another variable.

Reply:
I like that. Will turn it into an option that is set by the thread (not sent over a channel because we already share this variable) after past events have been handled.

Reply:
Didn't work out because then `update_with_events_forever` gets more complicated as it is possible that it gets an event from the stream first before past events.

The way you described would work but then we trade the atomic bool for a channel and channel code is more complicated than the bool in this case so I want to keep it as is.